grafana-ocp/templates/grafana.yaml

```diff
@@ -1,54 +1,50 @@
- 
+
 apiVersion: integreatly.org/v1alpha1
 kind: Grafana
 metadata:
   name: {{ .Values.name }}
   namespace: {{ .Release.Namespace }}
 spec:
+  client:
+    preferService: true
   containers:
-  - name: grafana-proxy
-    readinessProbe:
-      tcpSocket:
-        port: https    
-    env:
-      - name: HTTP_PROXY
-      - name: HTTPS_PROXY
-      - name: NO_PROXY
-    ports:
-      - name: https
-        containerPort: 3000
-        protocol: TCP
+  - name: oauth-proxy
+    image: openshift/oauth-proxy:latest
     imagePullPolicy: IfNotPresent
     volumeMounts:
-      - name: secret-grafana-tls
+      - name: secret-proxy-tls
         mountPath: /etc/tls/private
-      - name: secret-grafana-proxy
-        mountPath: /etc/proxy/secrets
-    image: quay.io/openshift/origin-oauth-proxy:4.3.0
+    ports:
+    - containerPort: 8443
+      name: public
     args:
-      - '-provider=openshift'
-      - '-https-address=:3000'
-      - '-http-address='
-      - '-email-domain=*'
-      - '-upstream=http://localhost:3001'
-      - '-openshift-sar={"resource": "namespaces", "verb": "get"}'
-      - '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get"}}'
-      - '-tls-cert=/etc/tls/private/tls.crt'
-      - '-tls-key=/etc/tls/private/tls.key'
-      - '-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token'
-      - '-cookie-secret-file=/etc/proxy/secrets/session_secret'
-      - '-openshift-service-account=grafana-serviceaccount'
-      - '-openshift-ca=/etc/pki/tls/cert.pem'
-      - '-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
-      - '-skip-auth-regex=^/metrics'
+      - --https-address=:8443
```

Comment:
this troubles me a bit because this basically means to give access to anyone who it authenticated. My approach actually required a specific permission. Can you comment the lines you don't need to we have them there for the future?

Reply:
added back the openshift-sar settings.